pkg/member/election_leader.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2023 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package member
+
+import (
+	"context"
+
+	"github.com/pingcap/kvproto/pkg/pdpb"
+	"github.com/pingcap/kvproto/pkg/tsopb"
+)
+
+// ElectionLeader defines the common interface of the leader, which is the pdpb.Member
+// for in PD/API service or the tsopb.Participant in the microserives.
+type ElectionLeader interface {
+	// GetListenUrls returns the listen urls
+	GetListenUrls() []string
+	// GetRevision the revision of the leader in etcd
+	GetRevision() int64
+	// String declares fmt.Stringer
+	String() string
+	// Watch on the leader in the election group
+	Watch(context.Context)
+}
+
+// EmbeddedEtcdLeader is the leader in the election group backed by the embedded etcd.
+type EmbeddedEtcdLeader struct {
+	parent   *EmbeddedEtcdMember
```

Comment:
How about renaming it? parent sounds strange

Reply:
parent indeed isn't a good name. Any suggestion for the name?

Reply:
container?

Reply:
candidate?

Reply:
wrapper?

Reply:
changed to wrapper

Reply:
SGTM

Reply:
change committed